File: tests/fixtures/dart-d5/positive/lib/state.dart
enum TypedPhase { queued, complete }

class Job {
  Job(this.state);

  String state;

  void queue() {
    state = 'queued';
  }

  void start() {
    state = 'running';
  }

  bool get isDone => 'done' == state;

  Map<String, Object> toJson() => {'state': state};
}

class TypedJob {
  TypedJob(this.phase);

  TypedPhase phase;
}

class LowEvidence {
  String status = 'queued';
}

String localStateDecoy() {
  var state = 'queued';
  state = 'running';
  return state;
}

class WirePayload {
  const WirePayload(this.state);

  final String state;

  Map<String, Object> toJson() => {'state': state};
}

void dynamicStateDecoy(dynamic payload) {
  payload.state = 'queued';
  if (payload.state == 'running') {
    payload.state = 'done';
  }
}

File: tests/fixtures/swift-semantic-a3/host/Sources/SwiftA3Core/State.swift
public enum JobState: String {
  case queued
  case running
  case done
}

public final class Job {
  private(set) var state: String

  public init() {
    state = "queued"
  }

  public func start() {
    state = "running"
  }

  public func finish() {
    state = "done"
  }

  public func isRunning() -> Bool {
    state == "running"
  }
}

public struct TypedJob {
  public var state: JobState

  public init(state: JobState) {
    self.state = state
  }
}

public struct WirePayload: Codable {
  public var status: String

  public init(status: String) {
    self.status = status
  }
}

public final class OneShot {
  private var phase = "new"

  public init() {}

  public func close() {
    phase = "closed"
  }
}
